### PR TITLE
ctest: add basic functional tests

### PR DESCRIPTION
**Pull request by ldorau**

### CODE REVIEW

File: test/test_basic.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+static void test_0(void)
+{
+	MARK_START();
+	run_basic_tests();
+	MARK_END();
+}
+
+static void test_1(void)
+{
+	(void) fork();
```
Comment (vitalyvch) [inline test/test_basic.c]: You definitely should do here **syscall(SYS_fork)**

**Review by vitalyvch — CHANGES_REQUESTED** [review]
Also please replace trace=kp-libc-all with trace=kp-kern-all

**Comment by ldorau** [comment]
> Also please replace trace=kp-libc-all with trace=kp-kern-all
Why?

**Comment by ldorau** [comment]
Fixed

File: test/test_basic.c (hunk 1 above)
Comment (ldorau) [reply test/test_basic.c]: Done

**Comment by ldorau** [comment]
Reopened in #3

**Pull request closed without merging**